src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -140,41 +140,188 @@ class F extends CopyToTextureUtils {
     return { canvas, canvasContext: gl };
   }
 
-  getExpectedPixels({
-    context,
+  getInitGPUCanvasData(
+    width: number,
+    height: number,
+    premultiplied: boolean,
+    paintOpaqueRects: boolean
+  ): Uint8ClampedArray {
+    const rectWidth = Math.floor(width / 2);
+    const rectHeight = Math.floor(height / 2);
+
+    const alphaValue = !premultiplied || paintOpaqueRects ? 255 : 153;
```

Comment:
Actually, I think there is a bug in chromium implementation for `GPUCanvasCompositingAlphaMode`. 
- When creating `StaticBitmapImage` from webgpu context canvas, the alpha attribute is not set correctly. The [code here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/webgpu/gpu_swap_chain.cc;l=161) always set the alpha mode to "premultiplied".
- The alpha value is not cleared to opaque after configuring `compositingAlphaMode = 'opaque'`
(According to [spec](https://gpuweb.github.io/gpuweb/#GPUCanvasCompositingAlphaMode), implementation should clear alpha to opaque(I think, 1.0/255 always))

This will fail the test when dstFormat is 16-32 bit (e.g. r16Float, r32Float) when `srcPremultipliedAlpha = false, dstPremultipliedAlpha = false`. 
Due to above implementation issues, the case will do an "unpremultiplied" op during copy and the value is not what we expected.

I plan to file an issue in chromium and fix it in my next sprint (next week).

And I think you're right, we don' need to workaround bug in cts. I'll change the logic here.

Reply:
Chromium still doesn't implement compositingAlphaMode at all:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/webgpu/gpu_canvas_configuration.idl

Also, it has yet to be determined yet whether compositingAlphaMode should affect using the canvas as a source or not:
https://github.com/gpuweb/gpuweb/issues/1847

Reply:
Ah, you're right. I misunderstand the code here https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/webgpu/gpu_canvas_context.cc;l=212 and think chromium already supported it.